### PR TITLE
GAT improvement: GAT to GATv2

### PR DESCRIPTION
**Pull request by pzhanggit**

### CODE REVIEW

File: models/GATStack.py [hunk 1]
```diff
@@ -16,8 +16,8 @@ def __init__(
         num_nodes: int,
         hidden_dim: int,
         config_heads: {},
-        heads: int = 1,
-        negative_slope: float = 0.2,
+        heads: int = 6,
```
Comment (allaffa) [inline models/GATStack.py:19]: @pzhanggit 
why are you setting as default a number of heads equal to 6? Shouldn't it remain 1?

File: models/GATStack.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def __init__(
         self.convs = ModuleList()
         self.batch_norms = ModuleList()
         self.convs.append(
-            GATConv(
+            GATv2Conv(
```
Comment (allaffa) [inline models/GATStack.py:36]: @pzhanggit @streeve 
I just read the description of the GATv2Conv layer in `torch.geometric`

File: test_graphs.py [hunk 3]
```diff
@@ -39,9 +39,9 @@ def pytest_train_model(model_type):
     # Set RMSE and sample error thresholds
     thresholds = {
         "PNN": [0.02, 0.10],
-        "MFC": [0.05, 0.15],
+        "MFC": [0.05, 0.20],
         "GIN": [0.08, 0.20],
-        "GAT": [0.10, 0.35],
+        "GAT": [0.05, 0.20],
```
Comment (allaffa) [inline test_graphs.py:44]: @pzhanggit 
Nice! :)

File: models/GATStack.py (hunk 1 above)
Comment (pzhanggit) [reply models/GATStack.py:19]: > @pzhanggit
> why are you setting as default a number of heads equal to 6? Shouldn't it remain 1?

To be more expressive. Also, 6 is the number found to allow the model to pass the unit test.

File: requirements-torchdep.txt [hunk 4]
```diff
@@ -1,3 +1,3 @@
-torch-geometric
+torch-geometric>=1.7.2
```
Comment (pzhanggit) [inline requirements-torchdep.txt:1]: Required for GATv2.

**Review by streeve — APPROVED** [review]

**Pull request merged**